File: 1018. Binary Prefix Divisible By 5.js
var prefixesDivBy5 = function(A) {
    let result = [];
    let previous = 0;
    for(let index = 0; index < A.length; index++){
        previous = (2*previous + A[index])%5;
        if(previous%5 == 0){
            result.push(true);
            continue;
        }
        result.push(false);
    }
    return result;
};

File: 1018. Binary Prefix Divisible By 5.java
class Solution {
    
    public List<Boolean> prefixesDivBy5(int[] A) {
        List<Boolean> result = new ArrayList();
        int previous = 0;
        for(int index = 0; index < A.length; index++){
            previous = (previous*2 + A[index])%5;
            if(previous % 5 == 0){
                result.add(true);
                continue;
            }
            result.add(false);
        }
        return result;
    }
}
